### ISSUE TITLE
Enhancement: Add title of @Ignore and @Issue annotations into report just before related test

### ISSUE THREAD

**Issue by rasimsitmanbetov** [renatoathaydes/spock-reports#39]
It would be nice to have titles of @Ignore and @Issue annotations in reports.

Reviewer will see the reason of a test failure or ignorance e.g. as a link to issue in bug tracker or some other resource.

For instance:
```
@Issue ("link_to_bug_tracker")
def "Failing test due to issue"() {
}

@Ignore ("Test isn't finished yet")
def "This could be a great test!"() {
}
```
Report should contain:
```
[some formatting related to @Issue] 	link_to_bug_tracker
[test_header_formatting] 				Failing test due to issue

[some formatting related to @Ignore] 	Test isn't finished yet
[test_header_formatting] 				This could be a great test!
```

**Comment by rdmueller** [comment]
+1 😄

**Comment by renatoathaydes** [comment]
Nice idea! Will try to do that soon!

**Comment by renatoathaydes** [comment]
I tried to implement this. Looks like we have a problem: Spock's `FeatureInfo` class has a `getReflection()` method which is used to get annotations on a feature. However, this method throws an `UnsupportedOperationException`, currently...

So I can see no way to get the value of the annotations so far.

**Comment by renatoathaydes** [comment]
Figured it out!

Need to do `FeatureInfo.description.getAnnotation( Ignore )`, not just `FeatureInfo.getAnnotation( Ignore )`. Weird...

**Comment by renatoathaydes** [comment]
The design for these two new features is in http://codepen.io/renatoathaydes/pen/ihGgt

Feature "Please ignore me" has an `@Ignore` annotation with value `Feature not implemented yet` and feature "A test with an error" has a `@Issue` annotation with value `Due to issue <a href='http://myhost.com/issue/998'>#998</a>`.

**Comment by renatoathaydes** [comment]
Consider this done. Will be released in 1.2.6.

**Comment by rasimsitmanbetov** [comment]
Sorry for late response!
Yes, the changes you did are exactly what was expected! Thanks!
